From 6b9fe865936db1c74cb811a3e98f9765d351f51f Mon Sep 17 00:00:00 2001
From: Qiu Wujie <b49553@freescale.com>
Date: Mon, 17 Nov 2014 14:06:29 +0800
Subject: [PATCH 29/45] mmc: sdio: make sure sdio_irq_thread pointer is not
 NULL.

Signed-off-by: Wujie Qiu <B49553@freescale.com>
Change-Id: Icf01605c71a19e0ffa01e85b1ed93eb0704493da
Reviewed-on: http://git.am.freescale.net:8181/23831
Tested-by: Review Code-CDREVIEW <CDREVIEW@freescale.com>
Reviewed-by: Xiaobo Xie <X.Xie@freescale.com>
Reviewed-by: Matthew Weigel <Matthew.Weigel@freescale.com>
[Original patch taken from QorIQ-SDK-V1.8-20150619-yocto]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 include/linux/mmc/host.h |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/include/linux/mmc/host.h b/include/linux/mmc/host.h
index 4ac5189..cb3f776 100644
--- a/include/linux/mmc/host.h
+++ b/include/linux/mmc/host.h
@@ -401,7 +401,8 @@ static inline void mmc_signal_sdio_irq(struct mmc_host *host)
 {
 	host->ops->enable_sdio_irq(host, 0);
 	host->sdio_irq_pending = true;
-	wake_up_process(host->sdio_irq_thread);
+	if (host->sdio_irq_thread)
+		wake_up_process(host->sdio_irq_thread);
 }
 
 void sdio_run_irqs(struct mmc_host *host);
-- 
1.7.5.4

